### PR TITLE
Rename adapter to repo

### PR DESCRIPTION
**Pull request by yordis**
I am following the footsteps of Ecto on this one, and be more clear about what we are trying to do.

We use adapter word in order to refer to the design pattern (I get that part), but we call it repo everywhere because that is what we are trying to swap using the Adapter pattern.

### CODE REVIEW

**Comment by yordis** [comment]
@alexfilatov please share some 1+

**Review by alexfilatov — COMMENTED** [review]
Yeah, I understand the change. But in this case if we took out behaviour from the project it will stop working. Maybe give this responsibility to smth else? @yordis

File: README.md [hunk 1]
```diff
@@ -25,7 +25,9 @@ end
 
 Then run `mix deps.get` on your terminal.
 
-Configure your application as seen in the *Configuration* section below prior to attempting to generate the migration or you will get an `application was not loaded/started` error.
+Configure your application as seen in the *Configuration* section below prior to
+attempting to generate the migration or you will get an
+`application was not loaded/started` error.
```
Comment (doomspork) [inline README.md:30]: Can we drop the changes to this file @yordis? We don't need to enforce a line limit in our markdown files.

**Review by doomspork — CHANGES_REQUESTED** [review]
I'm not sure I see the benefit of this PR @yordis, can you expand on the goal here?

Changing `Adapter` to `Repo` is the wrong change here I fear. Given that in Elixir most people are familiar with `Ecto.Repo` when using databases, calling this file in `Guardian.DB` `Repo` is _very_ confusing and it hides what the actual code does: provide an adapter. The original code refers to it as `repo` because it always was an `Ecto.Repo`, that is no longer the case.

When we look at this in code I find this to be very confusing:

```elixir
@behaviour Guardian.DB.Repo
```

Whereas this tells me we're using an adapter of some sort, not a derivative of `Ecto.Repo`:
```elixir
@behaviour Guardian.DB.Adapter
```
Furthermore, I'd like to avoid introducing things like 80 character line limits to markdown files as you've done in the README. This just introduces unnecessary change without provide any benefit.

File: lib/guardian/db.ex [hunk 2]
```diff
@@ -225,13 +225,6 @@ defmodule Guardian.DB do
     {:ok, amount_deleted}
   end
 
-  @doc false
-  def repo do
```
Comment (doomspork) [inline lib/guardian/db.ex:229]: Re-reading your PR it sounds like renaming this function and the config option to something other than `repo` would be the solution.

Adapter would suffice :+1:

**Pull request closed without merging**